designs/drift.md

```diff
@@ -0,0 +1,86 @@
+# Deprovisioning - Drift
+
+## Problem (v0.21.0 - v0.28.x)
+
+Provisioners and AWSNodeTemplates (AWS) are [declarative APIs](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#declarative-apis) that dictate the desired state of machines. User’s requirements for their machines as reflected in these CRDs can change over time. For example, users can add or remove labels or taints from their nodes, modify their instance requirements in their Provisioner, or modify the Subnets discovered by their AWSNodeTemplate. To enforce that requirements set in CRDs are applied to their fleet, users must manually terminate all out-of-spec nodes to rely on Karpenter to provision in-spec replacements.
```

Comment:
Can we reword to something like:

"Provisioners (e.g., `AWSNodeTemplates` in AWS) are..." and then use the general term (Provisioners) in the flow of the doc when we're talking about the CRD in general. If referencing a concrete CRD is helpful then we can use the `AWSNodeTemplate` and say so.

Does that make sense? Just laying the groundwork for future providers, and making it clear that in `karpenter-core` we're focused on the abstract thing (the Provisioner) rather than the concrete thing (`AWSNodeTemplate`).

Reply:
Yeah! Good callout. I go pretty extensively into some of the AWS API surfaces, thoughts on keeping these in the same doc? Or would you rather see them separately?

Reply:
I think using concrete examples is super helpful to understanding things when describing stuff like this, so we probably don't want to make things overly abstract. Just make it clear in the flow of the doc when we're using those concrete examples that it may not be exactly the same for your environment (in a theoretical world where `GCPNodeTemplates` or `AzureNodeTemplates` exist 😄).